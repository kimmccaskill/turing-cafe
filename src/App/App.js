import React, { Component } from 'react';
import './App.css';
import { getData } from '../apiCalls/apiCalls.js'
import Form from '../Form/Form.js';
import ResContainer from '../ResContainer/ResContainer.js';

class App extends Component {
  constructor() {
    super();
    this.state = {
      reservations: [],
    }
  }
  componentDidMount() {
    getData()
      .then(data => this.setState({reservations: data}))
  }

  render() {
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <Form />
        <ResContainer 
          reservations={this.state.reservations} 
        />
      </div>
    )
  }
}

export default App;
