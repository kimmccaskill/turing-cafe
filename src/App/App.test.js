import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('should be an instance of App component', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
