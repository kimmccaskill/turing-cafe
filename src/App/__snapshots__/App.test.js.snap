// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should be an instance of App component 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <Form
    addRes={[Function]}
  />
  <ResContainer
    reservations={Array []}
  />
</div>
`;
